credentials/credentials_test.go

```diff
@@ -0,0 +1,56 @@
+package credentials
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseL402Challenge(t *testing.T) {
+	tests := []struct {
+		name      string
+		challenge string
+		macaroon  string
+		invoice   string
+		expectErr string
+	}{
+		{
+			name:      "Invalid challenge: empty header",
+			challenge: "",
+			macaroon:  "",
+			invoice:   "",
+			expectErr: "no L402 challenge/empty header found",
+		},
+		{
+			name:      "Invalid challenge: missing macaroon",
+			challenge: "L402 invoice=1234",
+			macaroon:  "",
+			invoice:   "",
+			expectErr: "missing macaroon/invoice",
```

Comment:
is this the normal way in go, checking the actual error message string? is there anything like snapshots in JS ?

Reply:
I've seen in the other `credentials_test.go` that you actually match error type. What's the convention in go?

Reply:
there are some errors that are their own entity like "NotFound", those ones we can check them directly. However, `"missing macaroon/invoice"` is not one of those.

If you are catching/expecting to catch the returned error, then it means that is its own entity and you can use the error instead of the string.


For most of the errors, you can just use the string. If you change the message of an error it may make the test fail. However, that never happens/is solved in 2 min.